### PR TITLE
Vc patient registration

### PR DESCRIPTION
**Pull request by VikumChathuranga22434**
Completed the patient registration feature of the system

### CODE REVIEW

File: backend/routes/PRSS.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const router = express.Router();
+
+// import the patient function to the router
+const Patient = require("../systems/PRSS/patientReg");
+
+// Register routes here
+router.get("/test-patient", Patient.test);
```
Comment (yehan2002) [inline backend/routes/PRSS.js:8]: Remove test api

**Review by yehan2002 — CHANGES_REQUESTED** [review]

File: frontend/src/systems/PRS/components/PatientReg/PatientRegForm.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        nic: "",
+        phone: "",
+        gender: "",
+        email: "",
+        address: "",
+      });
+      setadobValue(dayjs(null)); // Reset the date picker value
+
+      // Refresh the page after submitting the form
+      window.location.reload();
```
Comment (yehan2002) [inline frontend/src/systems/PRS/components/PatientReg/PatientRegForm.tsx:90]: Reload page using react-router-dom https://stackoverflow.com/a/61123116/6587830

File: backend/systems/PRSS/patientReg.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+// localhost:5000/api/prss/test
+
+/**
+ * @param {express.Request} req
+ * @param {express.Response} res
+ */
+
+// test function
+async function test(req, res) {
```
[the remark this answers is not in the document]
Comment (VikumChathuranga22434) [reply backend/systems/PRSS/patientReg.js:42]: I think i need for some future implementations so after those done I'll remove it

File: backend/systems/PRSS/patientReg.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+// get patient details by id
+async function getPatientDetails(req, res) {
+  // check the patient id is null
+  if (!Number.isInteger(Number.parseInt(req.params.id))) {
+    res.status(400).json({ msg: "Invalid patient id" });
+    return;
+  }
+
+  const patient = await Patient.findByPk(req.params.id);
```
[the remark this answers is not in the document]
Comment (VikumChathuranga22434) [reply backend/systems/PRSS/patientReg.js]: I will look in to that

[the remark this answers is not in the document]
Comment (VikumChathuranga22434) [reply backend/systems/PRSS/patientReg.js]: Completed that issue

File: frontend/src/systems/PRS/components/PatientReg/MostRecentPatient.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  // patient details useState
+  const [recentPatientDetails, setRecentPatientDetails] = useState<any>();
+
+  // console log
+  console.log(recentPatientDetails);
+  // console.log(learnMore);
+
+  async function getRecentPatient() {
+    try {
+      const response = await fetch(`${BASE_URL}/api/prss/recent-patient`);
```
[the remark this answers is not in the document]
Comment (VikumChathuranga22434) [reply frontend/src/systems/PRS/components/PatientReg/MostRecentPatient.tsx:45]: After all done I'll close this issue @yehan2002

File: frontend/src/systems/PRS/components/PatientReg/PatientRegForm.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+          <TextField
+            fullWidth
+            id='phone'
+            label='Phone-No'
+            variant='standard'
+            margin='normal'
+            value={patient.phone}
+            onChange={handleInputChange}
+          />
+          <LocalizationProvider dateAdapter={AdapterDayjs}>
```
[the remark this answers is not in the document]
Comment (VikumChathuranga22434) [reply frontend/src/systems/PRS/components/PatientReg/PatientRegForm.tsx:162]: This issue yehan We will have a talk. need some information before delete it. And I need to check that this.

File: backend/routes/PRSS.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (VikumChathuranga22434) [reply backend/routes/PRSS.js:10]: I'll will do that after complete the patient registration. I already completed this feature but i think we need to have some improvement of the code so after looking in to that I'll delete this

File: frontend/src/systems/PRS/config.ts
Comment (yehan2002) [inline frontend/src/systems/PRS/config.ts:1]: Remove this file. Base path is already handled using axios.

**Review by yehan2002 — APPROVED** [review]

**Pull request merged**